src/components/QueryEditor/QueryHeader.tsx

```diff
@@ -44,6 +45,8 @@ const adxTimeFormat: SelectableValue<string> = {
 export const QueryHeader = (props: QueryEditorHeaderProps) => {
   const { query, onChange, schema, datasource, dirty, setDirty, onRunQuery, templateVariableOptions } = props;
   const { rawMode, OpenAI } = query;
+  const cluster = query.clusterUri;
```

Comment:
This should be a state variable I think. That would then allow you to conveniently update this in the `useEffect` if no cluster URI is specified.